src/main/java/org/openstreetmap/atlas/geography/atlas/statistics/coverage/poi/SimpleCoverage.java

```diff
@@ -0,0 +1,380 @@
+        filteredCoverages.forEach(definition ->
+        {
+            try
+            {
+                // Each line
+                final StringList split = StringList.split(definition, TYPE_SEPARATOR);
+                final StringList sources = StringList.split(split.get(1), VALUES_SEPARATOR);
+                final String type = split.get(0);
+                final String coverageTypes = split.size() > COVERAGE_TYPE_INDEX
+                        ? split.get(COVERAGE_TYPE_INDEX) : CoverageType.COUNT.name();
+                final Set<CoverageType> coverageTypeSet = StringList
+                        .split(coverageTypes, VALUES_SEPARATOR).stream().map(CoverageType::forName)
+                        .collect(Collectors.toSet());
+                final Set<TagGroup> allowedTags = new HashSet<>();
+                final StringList orGroups = StringList.split(split.get(2), KEYS_SEPARATOR);
+                orGroups.forEach(orGroup ->
+                {
+                    // Each "|" group
+                    final TagGroup tagGroup = new TagGroup();
+                    final StringList andGroups = StringList.split(orGroup, COUPLED_KEYS_SEPARATOR);
+                    andGroups.forEach(andGroup ->
+                    {
+                        final MultiMap<String, String> tags = new MultiMap<>();
+                        final StringList carets = StringList.split(andGroup, COUPLED_KEYS_GROUP);
+                        carets.forEach(caret ->
+                        {
+                            final StringList keyValue = StringList.split(caret,
+                                    KEY_VALUE_SEPARATOR);
+                            final String key = keyValue.get(0);
+                            final StringList values = StringList.split(keyValue.get(1),
+                                    VALUES_SEPARATOR);
+                            // Each "," possible value
+                            values.forEach(value -> tags.add(key, value));
+                        });
+                        tagGroup.addTags(tags);
+                    });
+                    allowedTags.add(tagGroup);
+                });
+                final BiFunction<String, CoverageType, SimpleCoverage<AtlasEntity>> simpleCoverageFunction = (
+                        metricName, sampleCoverageType) -> new SimpleCoverage<AtlasEntity>(
+                                LoggerFactory.getLogger(metricName), atlas, sampleCoverageType)
+                        {
+                            @Override
+                            protected Iterable<AtlasEntity> getEntities()
+                            {
+                                if (sources.contains("all"))
+                                {
+                                    return getAtlas();
+                                }
+                                final List<Iterable<? extends AtlasEntity>> result = new ArrayList<>();
+                                if (sources.contains("nodes"))
```

Comment:
Do you think it's worth pulling these constant out into an enum?

Reply:
I can do that..